runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -449,7 +460,12 @@ function M._enable(bufnr)
     api.nvim_create_autocmd('LspDetach', {
       buffer = bufnr,
       callback = function()
-        disable(bufnr)
+        -- Only disable if there are no more clients with diagnostics attached
+        -- LspDetach is called *before* the client disconnects, so the number of clients connected
+        -- to this buffer that support diagnostics will be 1 if we're the last client
+        if #vim.lsp.get_clients({ bufnr = bufnr, method = ms.textDocument_diagnostic }) == 1 then
```

Comment:
Hmm, this doesn't always seem to happen.

I think I need to check `vim.lsp.get_clients` and filter out the client that is detaching, and then see if any remain.